model/flow/transaction_body_builder.go

```diff
@@ -0,0 +1,289 @@
+package flow
+
+import (
+	"fmt"
+
+	"github.com/onflow/crypto"
+	"github.com/onflow/crypto/hash"
+	"golang.org/x/exp/slices"
+
+	"github.com/onflow/flow-go/model/fingerprint"
+)
+
+// TransactionBodyBuilder constructs a validated, immutable [TransactionBody] in two phases:
+// first by setting individual fields using fluent SetX methods, then by calling Build()
+// to perform minimal validity and sanity checks and return the final [TransactionBody].
+type TransactionBodyBuilder struct {
+	u UntrustedTransactionBody
+}
+
+// NewTransactionBodyBuilder help to build a new Transaction
+func NewTransactionBodyBuilder() *TransactionBodyBuilder {
+	return &TransactionBodyBuilder{}
+}
+
+// Build validates and returns an immutable TransactionBody. All required fields must be explicitly set (even if they are zero).
```

Comment:
```suggestion
// Build validates and returns an immutable TransactionBody.
```
I don't think this is enforced (which is OK)